<?xml version="1.0" encoding="UTF-8" ?>

<launch>

  <!-- PHASE 1: April tag publisher -->
  <arg name="april_node_namespace" default="apriltag_ros_continuous_node" />
  <rosparam command="load" file="$(find dafan_docking)/config/april_settings.yaml" ns="$(arg april_node_namespace)" />
  <rosparam command="load" file="$(find dafan_docking)/config/april_tags.yaml" ns="$(arg april_node_namespace)" />
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg april_node_namespace)" clear_params="true" output="screen" >
    <remap from="image_rect" to="/d435/color/image_raw" />
    <remap from="camera_info" to="/d435/color/camera_info" />
    <param name="camera_frame" type="str" value="d435_color_optical_frame" />
    <param name="publish_tag_detections_image" type="bool" value="false" /> 
  </node>
  <!-- PHASE 1: Run node to publish docking pose -->
  <node pkg="dafan_docking" name="april_docking" type="april_docking" output="screen" >
    <rosparam file="$(find dafan_docking)/config/phase1_april_docking.yaml" command="load"/>
  </node>

  <!-- PID Controller: Planner Adjuster PID Controller -->
  <node pkg="dafan_docking" type="dafan_planner_adjuster" name="dafan_planner_adjuster" output="screen">
    <rosparam file="$(find dafan_docking)/config/planner_adjuster.yaml" command="load" />
    <remap from="/cmd_vel" to="/cmd_vel_mux/autonomous"/>
  </node>

</launch>
